### ISSUE TITLE
BUG: Doc intersphinx link broken

### ISSUE THREAD

**Issue by larsoner** [dipy/dipy#2290]
## Description

In < 1.3.0, we could link to `dipy.align.imaffine.AffineMap`. In 1.3.0, this changes to `dipy.align._public.AffineMap`, which is not great:
```
$ python -m sphinx.ext.intersphinx https://dipy.org/documentation/1.2.0./objects.inv | grep " *dipy\.align.*AffineMap .*"
	dipy.align.imaffine.AffineMap            reference/dipy.align/#dipy.align.imaffine.AffineMap
$ python -m sphinx.ext.intersphinx https://dipy.org/documentation/1.3.0./objects.inv | grep " *dipy\.align.*AffineMap .*"
	dipy.align._public.AffineMap             reference/dipy.align/#dipy.align._public.AffineMap
```
And you can even see the `_public` in the name on the website:

https://dipy.org/documentation/1.3.0./reference/dipy.align/#affinemap

**Comment by skoudoro** [comment]
Indeed, it is not great, I do not know yet why this path got the priority over `dipy.align.imaffine.AffineMap`. 
Thank you for letting us know, we will look into it

**Comment by arokem** [comment]
I am not sure how this happened, but maybe we should implement `__all__` in dipy.align._public (and more generally).

**Comment by skoudoro** [comment]
> but maybe we should implement __all__ in dipy.align._public

👍

**Comment by Garyfallidis** [comment]
Thank you @larsoner. Great catch!

**Comment by larsoner** [comment]
Don't thank me, thank CircleCI and Sphinx with `-nW` nitpicky+warnings-as-error mode, they did the hard work :)

**Comment by skoudoro** [comment]
Just as a reminder and when we have time, we need to look at [this script](https://github.com/dipy/dipy/blob/master/doc/tools/apigen.py) to fix our issue with API doc and intersphinx. Our script seems to ignore `__all__`.

Look at #2291 for more info.
